### PR TITLE
Corrects invalid memory access in build_rir cython code

### PR DESCRIPTION
**Pull request by fakufaku**
Corrects invalid memory access in the ``pyroomacoustics.build_rir`` cython accelerator and adds a unit test that checks the cython code output is correct.

- [ ] Are there docstrings ? Do they follow the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard) style ?
- [X] Have you run the tests by doing `nosetests` or `py.test` at the root of the repo ?
- [ ] Have you checked that the doc builds properly and that any new file has been added to the repo ? How to do that is covered in the [documentation](https://pyroomacoustics.readthedocs.io/en/pypi-release/contributing.html#documentation).
- [X] Is there a unit test for the proposed code modification ? If the PR addresses an issue, the test should make sure the issue is fixed.
- [X] Last but not least, did you document the proposed change in the CHANGELOG file ? It should go under "Unreleased".

Comments:
* I'm not sure how to add docstring to cython extension code
* The doc stays unchanged

### CODE REVIEW

File: pyroomacoustics/tests/test_build_rir.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    
+
+if __name__ == '__main__':
+
+    import matplotlib.pyplot as plt
+
+    for t, a, v in zip(times, alphas, visibilities):
+        ir_ref, ir_cython = build_rir(times[0], alphas[0], visibilities[0], fs, fdl)
+
+        print('Error:', np.max(np.abs(ir_ref - ir_cython)))
```
Comment (ebezzam) [inline pyroomacoustics/tests/test_build_rir.py]: I get an error of `0.060633160336600614` when running the script. Is that expected or too high? I notice `tol=0.01`.

File: pyroomacoustics/build_rir.pyx [hunk 2]
```diff
@@ -41,9 +43,13 @@ def fast_rir_builder(
     fdl2 = (fdl - 1) // 2
     n_times = time.shape[0]
 
-    # create a look-up table of the since function and
+    assert time.shape[0] == visibility.shape[0]
+    assert time.shape[0] == alpha.shape[0]
+
```
[the remark this answers is not in the document]
Comment (fakufaku) [reply pyroomacoustics/build_rir.pyx]: Good thinking. Will do.

File: pyroomacoustics/tests/test_build_rir.py (hunk 1 above)
Comment (fakufaku) [reply pyroomacoustics/tests/test_build_rir.py]: That seems big... I wonder why this is the case... I get `0.001267`. What is your version of python / anaconda. Can you run a few more tests ? Also the continuous integration didn't fail, which means the output is within tolerance there as well...

Comment (fakufaku) [reply pyroomacoustics/tests/test_build_rir.py]: You can try to provide a larger `lut_gran` (20 is the default). This variable controls the number of points to use in the interpolation.

File: pyroomacoustics/build_rir.pyx [hunk 3]
```diff
@@ -41,9 +43,13 @@ def fast_rir_builder(
     fdl2 = (fdl - 1) // 2
     n_times = time.shape[0]
 
-    # create a look-up table of the since function and
```
[the remark this answers is not in the document]
Comment (fakufaku) [reply pyroomacoustics/build_rir.pyx]: Yup... 😓

File: pyroomacoustics/build_rir.pyx (hunk 2 above)
[the remark this answers is not in the document]
Comment (fakufaku) [reply pyroomacoustics/build_rir.pyx]: Done. Also checked for in the test.

File: pyroomacoustics/tests/test_build_rir.py (hunk 1 above)
Comment (ebezzam) [reply pyroomacoustics/tests/test_build_rir.py]: That (rebuild) was indeed the problem! Perhaps a "build from source" section can be added to the main README? I often forget the command (`python setup.py build_ext --inplace`
) and have to look into chat history 🙈 

I can make a separate PR for this if you want.

Comment (fakufaku) [reply pyroomacoustics/tests/test_build_rir.py]: We could add it to the workflow for the tests too. Feel free to do it! That'd be great!

Comment (fakufaku) [reply pyroomacoustics/tests/test_build_rir.py]: And thanks! I'll merge into master and make a release this week.

**Pull request merged**